tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Service/Microsoft.Legal.MatterCenter.Repository/Configs/ConfigRepository.cs

```diff
@@ -85,6 +93,94 @@ public List<DynamicTableEntity> GetConfigEntities(string filter)
+                Dictionary<string,string> keyValues = new Dictionary<string, string>();
+
+                Dictionary<string, EntityProperty> newProperties = new Dictionary<string, EntityProperty>();
+
+                //add all the configsGroups and their key value pairs to a dictionary
+                allValues = JsonConvert.DeserializeObject<Dictionary<string, Dictionary<string, string>>>(configs);
+
+                var batch = new TableBatchOperation();
+                TableQuery<DynamicTableEntity> queryFinal = new TableQuery<DynamicTableEntity>();
+       
+                foreach (KeyValuePair<string, Dictionary<string, string>> entry in allValues)
+                {                  
+                    foreach (KeyValuePair<string, string> keyValue in entry.Value)
+                    {                 
+                        TableQuery<DynamicTableEntity> entityQuery = new TableQuery<DynamicTableEntity>().Where(
+                            TableQuery.CombineFilters(
+                                TableQuery.GenerateFilterCondition("ConfigGroup", QueryComparisons.Equal.ToLower(), entry.Key),
+                                TableOperators.And,
+                                TableQuery.GenerateFilterCondition("Key", QueryComparisons.Equal, keyValue.Key)));
+
+                       IEnumerable<DynamicTableEntity> queryResult = table.ExecuteQuery(entityQuery);
+                       
+                        if (queryResult.GetEnumerator().MoveNext())
+                        {
+                            foreach (DynamicTableEntity entity in queryResult)                         
+                            {
+                                entity.Properties["ConfigGroup"].StringValue = entry.Key;
+                                entity.Properties["Key"].StringValue = keyValue.Key;
+                                entity.Properties["Value"].StringValue = keyValue.Value;
+                                batchOperation.Merge(entity);    
+                            }
+                        }
+                        else
+                        {
+                            DynamicTableEntity config = CreateEntity(entry, keyValue);
+                            batchOperation.InsertOrReplace(config);
+                        }                    
+                       
+                    }
+                }
+
+             table.ExecuteBatch(batchOperation);
+             return true;
+                         
+            }
+            catch (Exception ex)
+            {
+                throw;
+            }
+     
+
+        }
+
+        private DynamicTableEntity CreateEntity(KeyValuePair<string, Dictionary<string, string>> entry, KeyValuePair<string, string> keyValue)
+        {
+            DynamicTableEntity config = new DynamicTableEntity();
+            config.PartitionKey = "MatterCenterConfig";

```

Comment:
Check whether this can be configurable or not? If yes, read from appsettings.json file